src/bundle/Resources/public/js/alloyeditor/src/buttons/parts/ez-discover-content.js

```diff
@@ -0,0 +1,66 @@
+import React, { createRef } from 'react';
+import PropTypes from 'prop-types';
+import AlloyEditor from 'alloyeditor';
+import EzEmbedDiscoverContentButton from '../../base/ez-embeddiscovercontent';
+
+export default class EzDiscoverContent extends EzEmbedDiscoverContentButton {
+    constructor(props) {
+        super(props);
+
+        this.originalSetUIHidden = null;
+        this.buttonRef = createRef(null);
+    }
+
+    confirmSelection() {
+        const { confirmSelectedItems, editor } = this.props;
+
+        confirmSelectedItems(...arguments);
+
+        ReactDOM.unmountComponentAtNode(document.querySelector('#react-udw'));
+
+        editor._mainUI._setUIHidden = this.originalSetUIHidden;
+
+        this.buttonRef.current.focus();
+    }
+
+    cancelHandler(udwContainer) {
+        super.cancelHandler(udwContainer);
+
+        this.props.editor._mainUI._setUIHidden = this.originalSetUIHidden;
```

Comment:
Aren't you modyfing `this.props` here in some way? It looks strange and incorrect

Reply:
I know, I know. It is strange but this is needed for the AlloyEditor to stay open while we are discovering content in UDW.